feat: add option to use git-credentials file instead of token (#25336)
renovatebot/renovate: lib/modules/platform/gitlab/readme.md

- Make sure the user that owns the `RENOVATE_TOKEN` PAT is a member of the corresponding GitLab projects/groups with the right permissions.
- Make sure the `RENOVATE_TOKEN` PAT has the `read_registry` scope.

> If you are using a private gitlab and it is configured such as the HTTPS is configured for something else as HTTPS authentication (Kerberos authentication for example), you may want to use `git-credential` feature.

Reviewer: ```suggestion
If you are using a private gitlab and HTTPS is configured for something else  like Kerberos authentication, you may want to use `platformGitCredentialsFile` feature.
```

> It will not send the token in the URL but rather write the `.git-credentials` file (if it doesn't exist) and configure `git` to use it.
> You can activate this feature by setting [`platformGitCredentialsFile`](https://docs.renovatebot.com/configuration-options/#platformGitCredentialsFile) to `true`.

Reviewer: ```suggestion
```
